### PR TITLE
Translate documentation files (PT-BR): Type Inference / Type Compatibility 

### PR DESCRIPTION
**Pull request by Perkles**
@khaosdoctor  @danilofuchs @orta 

This PR (Ref #233) translates the following files to Brazilian Portuguese:
 - Type Inference
 - Type Compatibility 

During the translation process I found the following title: 

### [A note about Soundness](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/documentation/copy/en/reference/Type%20Compatibility.md#a-note-on-soundness)

Wich introduced the therm `Soundness` into the text. According to Wikipedia Soundness could be interpreted as:

> [In logic, more precisely in deductive reasoning, an argument is sound if it is both valid in form and its premises are true.[1] Soundness also has a related meaning in mathematical logic, wherein logical systems are sound if and only if every formula that can be proved in the system is logically valid with respect to the semantics of the system.](https://en.wikipedia.org/wiki/Soundness)

Soundness is often applied as "sound" and "unsound" throughout the text. I'd had trouble finding a corresponding word to use so I stick with the Wikipedia [Portuguese translation](https://pt.wikipedia.org/wiki/Corre%C3%A7%C3%A3o) and used:
 - Soundness as 'Correção'
 - Sound as 'Correto'
 - Unsound as 'Incorreto'

### CODE REVIEW

**Comment by khaosdoctor** [comment]
@perkles we have another PR on soundness, which has been translated to "solidez" (check #1212)

File: packages/documentation/copy/pt/reference/Type Compatibility.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+---
+title: Type Compatibility
+layout: docs
+permalink: /docs/handbook/type-compatibility.html
+oneline: How type-checking works in TypeScript
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Type Compatibility.md]: You can translate this line too

File: packages/documentation/copy/pt/reference/Type Compatibility.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+  nome: string;
+}
+
+let p: Nomeado;
+// OK, por causa da tipagem estrutural
+p = new Pessoa();
+```
+Em linguagens nominalmente tipadas como C# ou Java, o código equivalente seria um erro porque a classe `Pessoa` não se descreve explícitamente como sendo um implementador da interface `Nomeado`.
+
+O sistema de tipagem estrutura do TypeScript foi projetado baseado em como o código Javascript é tipicamente escrito.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Type Compatibility.md]:
```suggestion
O sistema de tipagem estrutural do TypeScript foi projetado baseado em como o código Javascript é tipicamente escrito.
```

File: packages/documentation/copy/pt/reference/Type Compatibility.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+Por que o Javascript faz uso amplo de objetos anônimos como funções de expressões e literais de objetos, é muito mais natural representar os tipos de relacionamentos encontrados em bibliotecas JavaScript com um sistema de tipo estrutural do que um tipo nominal.
+
+## Uma nota sobre Correção
+
+O sistema de tipos do TypeScript permite certas operações, que não poderiam ser conhecidas em tempo de compilação, serem seguras.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Type Compatibility.md]:
```suggestion
O sistema de tipos do TypeScript permite certas operações, que não poderiam ser conhecidas em tempo de compilação, a serem seguras.
```

File: packages/documentation/copy/pt/reference/Type Compatibility.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+Quando um sistema de tipos tem essa propriedade, fica dito para não ser "correto". Os locais onde Typescript permite comportamento incorreto foram considerados cuidadosamente, e ao decorrer desse documento iremos explicar onde eles ocorrem e os cenários motivadores por tás deles.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Type Compatibility.md]:
```suggestion
Quando um sistema de tipos tem essa propriedade, fica dito para não ser "Sólido" (_Sound_). Os locais onde Typescript permite comportamento incorreto foram considerados cuidadosamente, e ao decorrer desse documento iremos explicar onde eles ocorrem e os cenários motivadores por trás deles. 
```

File: packages/documentation/copy/pt/reference/Type Compatibility.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+```
+
+Perceba que `y` tem uma propriedade `localizacao` extra, mas isso não cria um erro.
+Apenas membros do tipo alvo (`Nomeado` nesse caso) são considerados ao checar por compatibilidade.
+
+Esse processo de comparação ocorre recursivamente, explorando o tipo de cada membro e sub-membro.
+
+## Comparando duas funções
+
+Enquanto comparar tipos primitivos é relativamente simples, a questão de quais tipos de funções deveriam ser consideradas compatíveis é um pouco mais embaralhado. Vamos começar com um exemplo básico de duas funções que diferem apenas em sua lista de parâmetros:
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Type Compatibility.md]:
```suggestion
Enquanto comparar tipos primitivos é relativamente simples, a questão de quais tipos de funções deveriam ser consideradas compatíveis é um pouco mais difícil. Vamos começar com um exemplo básico de duas funções que diferem apenas em sua lista de parâmetros:
```

File: packages/documentation/copy/pt/reference/Type Inference.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
+---
+title: Type Inference
+layout: docs
+permalink: /docs/handbook/type-inference.html
+oneline: How code flow analysis works in TypeScript
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Type Inference.md]: You can translate this part too :)

**Review by khaosdoctor — CHANGES_REQUESTED** [review]

**Comment by Perkles** [comment]
Solidez fits perfectly 😄 
I've used:
 - Soundness as "Solidez"
 - Sound as "Sólido"
 - Unsound as "Não Sólido"

**Comment by orta** [comment]
The CI fail is legit BTW:

```
➤ YN0000: [documentation]: /home/runner/work/TypeScript-Website/TypeScript-Website/packages/documentation/scripts/generateDocsNavigationPerLanguage.js:334
➤ YN0000: [documentation]:     throw new Error(
➤ YN0000: [documentation]:     ^
➤ YN0000: [documentation]: 
➤ YN0000: [documentation]: Error: Permalink in /home/runner/work/TypeScript-Website/TypeScript-Website/packages/documentation/copy/pt/reference/Type Compatibility.md does not start with '/pt/'
```

**Review by khaosdoctor — APPROVED** [review]
Nice!

**Comment by khaosdoctor** [comment]
lgtm

**Pull request merged**